src/Driver/SolverTypes/ImplicitSolverType.jl

```diff
@@ -0,0 +1,69 @@
+
+export ImplicitSolverType
+
+"""
+# Description
+    ImplicitSolverType(;
+        solver_method = KenCarp4,
+    )
+
+This solver type constructs an ODE solver using an implicit
+method.
+
+# Arguments
+- `solver_method` (Function): Function defining the implicit
+    solver.
+    Default: `KenCarp4`
+"""
+struct ImplicitSolverType <: AbstractSolverType
+    # Function for an implicit method
+    solver_method::Function
+
+    function ImplicitSolverType(
+        alg;
+        solver_method = ODESolvers.DiffEqJLConstructor(alg),
+    )
+
+        return new(solver_method)
+    end
+end
+
+"""
+    getdtmodel(ode_solver::AbstractSolverType, bl)
+
+A function which returns a model representing the dynamics
+with the most restrictive time-stepping requirements.
+"""
+function getdtmodel(ode_solver::ImplicitSolverType, bl)
+    # For implicit methods, the entire model itself
+    # contributes to the total stability of the time-integrator
+    return getdt(ode_solver.solver_method)
```

Comment:
This should return a balance law. This is ultimately used in the `calculate_dt` function in the main driver.